db/data_migration/20230104104917_reslug_her_majesty_to_his_majesty.rb

```diff
@@ -0,0 +1,105 @@
+require "csv"
+require "uri"
+
+csv_file = File.join(File.dirname(__FILE__), "20230104104916_reslug_her_majesty_to_his_majesty.csv")
+csv = CSV.parse(File.open(csv_file), headers: true)
+
+csv.each do |row|
+  puts "reslugging #{row['current_url']}"
+
+  old_slug = URI(row["current_url"]).path.split("/").last
+  new_slug = URI(row["new_url"]).path.split("/").last
+
+  case row["content_type"]
+  when "html_publication"
+    publication_slug = URI(row["current_url"]).path.split("/").second_to_last
+    documents = Document.where(slug: publication_slug)
+    if documents.count > 1
+      puts "There are multiple documents with the slug '#{publication_slug}'. Consider writing a migration and fetching the document with a content_id or document_type to uniquely identify it."
+      next
+    end
+
+    document = documents.first
+    edition = document&.editions&.published&.last
+    unless edition
+      puts "!! edition not found for: #{old_slug}"
+      next
+    end
+
+    html_attachment = edition.attachments.find_by(slug: old_slug)
+
+    unless html_attachment
+      puts "!! no HTML attachments found for: #{old_slug}"
+      next
+    end
+
+    # update slug of html attachement and add redirect
+    html_attachment.update!(slug: new_slug)
+    Whitehall::PublishingApi.republish_async(html_attachment)
+
+    # remove the most recent edition from the search index
+    Whitehall::SearchIndex.delete(edition)
+
+    # send edition to publishing api
+    PublishingApiDocumentRepublishingWorker.new.perform(document.id)
+
+    # add edition to search index
+    Whitehall::SearchIndex.add(edition)
+
+  when "official_statistics_announcement"
+    statistics_announcement = StatisticsAnnouncement.find_by(slug: old_slug)
+
+    unless statistics_announcement
+      puts "!! no statistics announcement found for: #{old_slug}"
+      next
+    end
+
+    Whitehall::SearchIndex.delete(statistics_announcement)
+    statistics_announcement.update!(slug: new_slug)
+    Whitehall::SearchIndex.add(statistics_announcement)
```

Comment:
Double check: does this need to be sent to the Publishing API?

**I think it probably doesn't** – because StatisticsAnnouncement includes PublishesToPublishingApi, which will publish to Publishing API after committing to the database.

https://github.com/alphagov/whitehall/blob/c465d1d5a806d31333a7b2f01461f0a38ee59461/lib/publishes_to_publishing_api.rb#L6

But worth double checking in case our Friday brains are failing us.